azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/SqlServer.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+package com.microsoft.azure.toolkit.lib.sqlserver;
+
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.resources.fluentcore.arm.ResourceId;
+import com.azure.resourcemanager.sql.SqlServerManager;
+import com.microsoft.azure.toolkit.lib.common.entity.AbstractAzureEntityManager;
+import com.microsoft.azure.toolkit.lib.common.entity.IAzureEntityManager;
+import com.microsoft.azure.toolkit.lib.common.event.AzureOperationEvent;
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import com.microsoft.azure.toolkit.lib.common.operation.AzureOperation;
+import com.microsoft.azure.toolkit.lib.common.utils.NetUtils;
+import com.microsoft.azure.toolkit.lib.database.IDatabaseServerUpdater;
+import com.microsoft.azure.toolkit.lib.database.JdbcUrl;
+import com.microsoft.azure.toolkit.lib.database.entity.FirewallRuleEntity;
+import com.microsoft.azure.toolkit.lib.sqlserver.model.SqlDatabaseEntity;
+import com.microsoft.azure.toolkit.lib.sqlserver.model.SqlServerEntity;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+
+import javax.annotation.Nonnull;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SqlServer extends AbstractAzureEntityManager<SqlServer, SqlServerEntity, com.azure.resourcemanager.sql.models.SqlServer>
+        implements AzureOperationEvent.Source<SqlServer>, IAzureEntityManager<SqlServerEntity> {
+
+    private final SqlServerManager manager;
+
+    public SqlServer(@Nonnull SqlServerManager manager, @Nonnull com.azure.resourcemanager.sql.models.SqlServer server) {
+        super(new SqlServerEntity(server));
+        this.manager = manager;
+    }
+
+    @Override
+    protected com.azure.resourcemanager.sql.models.SqlServer loadRemote() {
+        try {
+            this.entity().setRemote(manager.sqlServers().getById(this.entity.getId()));
```

Comment:
I think the setRemote shall be handled in AbstractAzureEntityManager

Reply:
updated